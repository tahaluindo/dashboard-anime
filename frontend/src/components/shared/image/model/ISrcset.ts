export interface ISrcset {
  minimal: string,
  medium: string,
  large: string,
  huge: string,
  default: string,
}

