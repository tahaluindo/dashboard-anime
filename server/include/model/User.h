#include <string>

#ifndef USER_H
#define USER_H

struct User
{
  std::string cardId;
  std::string name;
};

#endif
