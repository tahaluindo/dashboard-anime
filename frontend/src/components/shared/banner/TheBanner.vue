<template>
  <div ref="banner" class="relative w-full overflow-hidden h-45vw">
    <the-image-background :srcset="srcset" :alt="alt || 'banner-background'" />
    <slot />
  </div>
</template>

<script lang="ts">
import { defineComponent, onMounted, PropType, ref } from "vue";
import { ISrcset } from "../image/model/ISrcset";
import { HeaderObserverKey } from "../header/controllers/HeaderObserver";
import TheImageBackground from "../image/TheImageBackground.vue";
import { injectStrict } from "../../../helpers/utils";

export default defineComponent({
  components: { TheImageBackground },
  props: {
    srcset: {
      type: Object as PropType<ISrcset>,
      required: true,
    },
    alt: {
      type: String,
    },
  },
  setup() {
    const banner = ref(null as unknown as HTMLElement);
    const headerObserver = injectStrict(HeaderObserverKey);

    onMounted(() => {
      headerObserver.observe(banner.value);
    });

    return {
      banner,
    };
  },
});
</script>

<style lang="postcss" scoped></style>
