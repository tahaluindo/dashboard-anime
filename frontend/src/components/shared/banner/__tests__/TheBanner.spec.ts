import { config, mount } from "@vue/test-utils";
import { mockWindowIntersectionObserver } from "../../../../__mocks__/window.mock";
import {
  HeaderObserver,
  HeaderObserverKey,
} from "../../header/controllers/HeaderObserver";
import { ISrcset } from "../../image/model/ISrcset";
import TheBanner from "../TheBanner.vue";

describe("TheBanner.vue", () => {
  let headerObserver: HeaderObserver;

  beforeEach(() => {
    mockWindowIntersectionObserver();

    headerObserver = new HeaderObserver();
  });

  beforeAll(() => {
    config.renderStubDefaultSlot = true;
  });

  afterAll(() => {
    config.renderStubDefaultSlot = false;
  });

  it("renders the default component with the props passed", () => {
    const srcset: ISrcset = {
      minimal: "minimal",
      medium: "medium",
      large: "large",
      huge: "huge",
      default: "default",
    };
    const wrapper = mount(TheBanner, {
      props: {
        srcset: srcset,
      },
      global: {
        provide: {
          [HeaderObserverKey as symbol]: headerObserver,
        },
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
