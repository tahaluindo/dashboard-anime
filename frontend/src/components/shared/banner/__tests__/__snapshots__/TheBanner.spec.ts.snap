// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheBanner.vue renders the default component with the props passed 1`] = `
"<div class=\\"relative w-full overflow-hidden h-45vw\\">
  <div class=\\"absolute inset-0 z-0\\">
    <picture>
      <source srcset=\\"minimal\\" media=\\"(max-width: 800px)\\" width=\\"800\\" height=\\"400\\">
      <source srcset=\\"medium\\" media=\\"(max-width: 1000px)\\" width=\\"1000\\" height=\\"450\\">
      <source srcset=\\"large\\" media=\\"(max-width: 1200px)\\" width=\\"1200\\" height=\\"500\\">
      <source srcset=\\"huge\\" media=\\"(min-width: 1200px)\\" width=\\"1200\\" height=\\"500\\"><img class=\\"object-cover w-full min-h-full\\" src=\\"default\\" alt=\\"banner\\">
    </picture>
    <!--v-if-->
  </div>
</div>"
`;
